research/training/tokenizer/train.py

```
@@ -419,6 +481,78 @@ def train(cfg: TrainConfig) -> dict:
+            for k, v in metrics.items()
+            if isinstance(v, (int, float)) and not isinstance(v, bool)
+        ]
+        snapshot = TrainingRunEvalSnapshot(
+            modality="image",
+            dataset=TrainingRunEvalDataset(name=ds_name, split=split, sha256="pending"),
+            step=step,
+            generatedAt=utc_now_iso(),
+            metrics=metric_objs,
+        )
+        return [snapshot]
+    except Exception as exc:  # eval is best-effort; never fail the run
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)